src/ng/cookieReader.js

```diff
@@ -22,6 +22,13 @@ function $$CookieReader($document) {
     }
   }
 
+  //null object implementation in case document was mocked out
+  if (!rawDocument || typeof rawDocument.cookie !== 'string') {
+    return function() {
+      return lastCookies;
+    };
+  }
+
```

Comment:
I think this is probably overkill. If the developer mocks out window and doesn't include a document then lots of stuff is going to break. Instead, how about we simply do

```js
lastCookieString = rawDocument.cookie || '';
```

In the function below.